cmd/scollector/collectors/nexpose.go

```diff
@@ -0,0 +1,405 @@
+package collectors
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"bosun.org/cmd/scollector/conf"
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
+)
+
+type loginRequest struct {
+	XMLName  xml.Name `xml:"LoginRequest"`
+	UserID   string   `xml:"user-id,attr"`
+	Password string   `xml:"password,attr"`
+}
+
+type apiResponse struct {
+	XMLName       xml.Name
+	Message       string        `xml:"Message"`
+	SessionID     string        `xml:"session-id,attr"`
+	SiteSummaries []siteSummary `xml:"SiteSummary"`
+	ScanSummaries []scanSummary `xml:"ScanSummary"`
+	Devices       []device      `xml:"SiteDevices>device"`
+}
+
+type simpleRequest struct {
+	XMLName   xml.Name
+	SessionID string `xml:"session-id,attr"`
+}
+
+type siteRequest struct {
+	XMLName   xml.Name
+	SessionID string `xml:"session-id,attr"`
+	SiteID    int    `xml:"site-id,attr"`
+}
+
+type siteSummary struct {
+	ID           int     `xml:"id,attr"`
+	Name         string  `xml:"name,attr"`
+	Description  string  `xml:"description,attr"`
+	RiskFactor   float32 `xml:"riskfactor,attr"`
+	RiskScore    float64 `xml:"riskscore,attr"`
+	ExploitCount int
+	MalwareCount int
+	VulnCount    int
+}
+
+type device struct {
+	ID         int     `xml:"id,attr"`
+	Address    string  `xml:"address,attr"`
+	RiskFactor float32 `xml:"riskfactor,attr"`
+	RiskScore  float64 `xml:"riskscore,attr"`
+}
+
+type scanSummary struct {
+	ID        int    `xml:"scan-id,attr"`
+	SiteID    int    `xml:"site-id,attr"`
+	EngineID  int    `xml:"engine-id,attr"`
+	Name      string `xml:"name,attr"`
+	StartTime string `xml:"startTime,attr"`
+	EndTime   string `xml:"endTime,attr"`
+	Status    string `xml:"status,attr"`
+}
+
+func init() {
+	registerInit(func(c *conf.Conf) {
+		for _, n := range c.Nexpose {
+			collectors = append(collectors, &IntervalCollector{
+				F: func() (opentsdb.MultiDataPoint, error) {
+					return c_nexpose(n.Username, n.Password, n.Host, n.Insecure)
+				},
+				name: fmt.Sprintf("nexpose-%s", n.Host),
+			})
+		}
+	})
+}
+
+type nexposeConnection struct {
+	Username  string
+	Password  string
+	Host      string
+	URLv1     string
+	URLv2     string
+	Client    *http.Client
+	SessionID string
+	Cookie    *http.Cookie
+}
+
+func c_nexpose(username, password, host string, insecure bool) (opentsdb.MultiDataPoint, error) {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: insecure},
```

Comment:
Just noting the irony of even having an insecure option for a vulnerability tool.